k8s/resolver.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Walks a yaml object and resolves all docker_name.Tag to docker_name.Digest.
+"""
+
+import argparse
+import sys
+
+from containerregistry.client import docker_creds
+from containerregistry.client import docker_name
+from containerregistry.client.v2_2 import docker_image as v2_2_image
+from containerregistry.tools import patched
+from containerregistry.transport import transport_pool
+
+import httplib2
+import yaml
+
+
+parser = argparse.ArgumentParser(
+    description='Resolve image references to digests.')
+
+parser.add_argument(
+  '--template', action='store',
+  help='The template file to resolve.')
+
+
+def Resolve(input, tag_to_digest):
+  """Translate tag references within the input yaml into digests."""
```

Comment:
So the idea is that *all* image references in a yaml get resolved into a digest? Is there a way to specify just a subset of them to resolve? I guess it's not a big deal, but I'm thinking of cases where users specify sidecar containers maintained by other teams/projects.

Reply:
Yeah, it resolves everything.  Once I add `images={}` it may make sense to also add an option that will tell this to *only* resolve published images, but I think my preference is just that: opt out.